### DISCUSSION TITLE
What should be use to get the list of markdown files in a directory?

### DISCUSSION THREAD

**Discussion by JeremieLitzler** [vuepress/core#380] [Q&A] [answered]
Hello, 

Thanks @meteorlxy, I was able to understand the registration of a compoment.
I used the manual registration through clientAppEnhance.

However, as I read the migration guide, I saw `$site.pages` was also dropped for scalability issues (#2689).

So, now:

- How can we now list the markdown files in a directory from a component?
- How can we access the frontmatter of each file?

When I look at VuePress > Page, I bet it is available but how?

I tried to understand the Client API but I am getting for `Error: useSiteLocaleData() is called without provider.`

My goal: build an blog index of the markdown files in a given directory.

- Link to repo: 
  - See [source component](https://github.com/JeremieLitzler/journal/tree/main/docs/.vuepress/components).
  - See [source markdown](https://github.com/JeremieLitzler/journal/blob/main/docs/blog/README.md) or [raw version](https://raw.githubusercontent.com/JeremieLitzler/journal/main/docs/blog/README.md).

I am still a little novice with Vue and Node development.

Thanks for the time taken to answer and helping me learn :)

**Comment by meteorlxy** [comment, marked as answer]
See #72

**Reply by JeremieLitzler** [reply]
Thanks @meteorlxy . I shall look it up later.
